tests/Core/Ruleset/DisplayCachedMessagesTest.php

```diff
@@ -0,0 +1,306 @@
+
+
+    /**
+     * Test that non-blocking messsages will not show when specific command-line options are being used.
+     *
+     * @param array<string> $configArgs Arguments to pass to the Config.
+     *
+     * @dataProvider dataSelectiveDisplayOfMessages
+     *
+     * @return void
+     */
+    public function testNonBlockingErrorsDoNotShowUnderSpecificCircumstances($configArgs)
+    {
+        $config  = new ConfigDouble($configArgs);
+        $ruleset = new Ruleset($config);
+        $this->mockCachedMessages($ruleset, ['Deprecation notice' => MsgCollector::DEPRECATED]);
+
+        $this->expectOutputString('');
+
+        $this->invokeDisplayCachedMessages($ruleset);
+
+    }//end testNonBlockingErrorsDoNotShowUnderSpecificCircumstances()
+
+
+    /**
+     * Data provider.
+     *
+     * @see testBlockingErrorsAlwaysShow()
+     * @see testNonBlockingErrorsDoNotShow()
+     *
+     * @return array<string, array<string, string|array<string>>>
+     */
+    public static function dataSelectiveDisplayOfMessages()
+    {
+        return [
+            'Explain mode'               => [
+                'configArgs' => ['-e'],
+            ],
+            'Quiet mode'                 => [
+                'configArgs' => ['-q'],
+            ],
+            'Documentation is requested' => [
+                'configArgs' => ['--generator=text'],
+            ],
+        ];
+
+    }//end dataSelectiveDisplayOfMessages()
+
+
+    /**
+     * Test Helper.
+     *
+     * @return \PHP_CodeSniffer\Ruleset
+     */
+    private function getPlainRuleset()
+    {
+        static $ruleset;
+
+        if (isset($ruleset) === false) {
+            $config  = new ConfigDouble();
+            $ruleset = new Ruleset($config);
+        }
+
+        return $ruleset;
+
+    }//end getPlainRuleset()
+
+
+    /**
+     * Add mock messages to the message cache.
+     *
+     * @param \PHP_CodeSniffer\Ruleset $ruleset  The ruleset object.
+     * @param array<string, int>       $messages The messages to add to the message cache.
+     *
+     * @return void
+     */
+    private function mockCachedMessages(Ruleset $ruleset, $messages)
+    {
+        $reflProperty = new ReflectionProperty($ruleset, 'msgCache');
+        $reflProperty->setAccessible(true);
+
+        $msgCache = $reflProperty->getValue($ruleset);
+        foreach ($messages as $msg => $type) {
+            $msgCache->add($msg, $type);
+        }
+
+        $reflProperty->setAccessible(false);
+
+    }//end mockCachedMessages()
+
+
+    /**
+     * Invoke the display of the cached messages.
+     *
+     * @param \PHP_CodeSniffer\Ruleset $ruleset The ruleset object.
+     *
+     * @return void
+     */
+    private function invokeDisplayCachedMessages(Ruleset $ruleset)
+    {
+        $reflMethod = new ReflectionMethod($ruleset, 'displayCachedMessages');
+        $reflMethod->setAccessible(true);
+        $reflMethod->invoke($ruleset);
+        $reflMethod->setAccessible(false);
```

Comment:
Do we need to set this "back" to accessible=false? The object (`$reflMethod`) will be cease to exist immediately after this line of code.

```suggestion
```

Reply:
Force of habit just in case ;-)